api/apps/api/test/implicit-permissions/implicit-permissions.e2e-spec.ts

```diff
@@ -0,0 +1,97 @@
+import { FixtureType } from '@marxan/utils/tests/fixture-type';
+import { getFixtures } from './implicit-permissions.fixtures';
+
+let fixtures: FixtureType<typeof getFixtures>;
+
+beforeEach(async () => {
+  fixtures = await getFixtures();
+});
+afterEach(async () => {
+  await fixtures?.cleanup();
+});
+
+test(`All users roles in project must be replicated in newly created scenario`, async () => {
+  /*
+    To test everything at once and for clarity purposes, we will
+    consider the creator of project not as owner but as "creator",
+    as another owner will be added as well to test that every single
+    role is transferred to a scenario regardless of their status
+    as creator or not. The owner that does not create the project
+    would be mentioned just as "owner" or "other owner".
+  */
+  await fixtures.GivenProjectWasCreated();
+  const creatorToken = await fixtures.GivenUserIsLoggedIn('owner');
+  await fixtures.GivenOwnerWasAddedToProject();
+  await fixtures.GivenContributorWasAddedToProject();
+  await fixtures.GivenViewerWasAddedToProject();
+
+  const scenarioId = await fixtures.GivenScenarioWasCreated();
+
+  const response = await fixtures.WhenGettingScenarioUsers(
+    scenarioId,
+    creatorToken,
+  );
+
+  fixtures.ThenAllUsersInProjectAreReturnedWithTheProperRoleInScenario(
+    response,
+  );
+});
+
+test(`All users roles in scenario should be revoked when revoked in project`, async () => {
+  await fixtures.GivenProjectWasCreated();
+  const creatorToken = await fixtures.GivenUserIsLoggedIn('owner');
+
+  await fixtures.GivenOwnerWasAddedToProject();
+  await fixtures.GivenContributorWasAddedToProject();
+  await fixtures.GivenViewerWasAddedToProject();
+
+  const scenarioId = await fixtures.GivenScenarioWasCreated();
+
+  await fixtures.WhenDeletingAllUsersExceptCreatorFromProject();
+
+  const response = await fixtures.WhenGettingScenarioUsers(
+    scenarioId,
+    creatorToken,
+  );
+  fixtures.ThenOnlyCreatorIsReturned(response);
+});
+
+test(`All changes in roles at project level should also be done at scenario level`, async () => {
+  await fixtures.GivenProjectWasCreated();
+  const creatorToken = await fixtures.GivenUserIsLoggedIn('owner');
+
+  await fixtures.GivenOwnerWasAddedToProject();
+  await fixtures.GivenContributorWasAddedToProject();
+  await fixtures.GivenViewerWasAddedToProject();
+
+  const scenarioId = await fixtures.GivenScenarioWasCreated();
+
+  await fixtures.WhenChangingAllUsersRole();
+  const response = await fixtures.WhenGettingScenarioUsers(
+    scenarioId,
+    creatorToken,
+  );
+  fixtures.ThenUsersWithChangedRolesShouldBeReturned(response);
+});
+
+test(`getting all locks of scenarios as scenario owner`, async () => {
```

Comment:
I realized that a couple of tests do not run properly. I think there were some problems on how I approached the tests and I made some changes because maybe they were the reason.

I also introduced a DELETE FROM into the manage_existing_projects SQL trigger function, because when testing locally I found some weird messages appearing, but If you consider it is not necessary just delete it.

If tests pass and you're happy with them, I think this can be merged, as I don't want it to take any more time of my life. Locally I got it working by applying the actions manually. If tests keep failing maybe I overlooked something in how the fixtures are built or something.